docs/1-getting-started/creating-a-window.md

```diff
@@ -1 +1,52 @@
 # Creating a Window
+
+Creating a window using the Windows API library (a.k.a WINAPI32) can be made by following these steps:"
+
+## 1.) Fill a window class structure (WNDCLASS)
+
+```cpp
+WNDCLASS wndClass = {};
+wndClass.hInstance = hInstance; //hInstance from wWinMain/WinMain
+wndClass.lpfnWndProc = DefWindowProc; //function used to handle messages. We're currently using the default window message handling procedure. We will create our message handler function later...
+wndClass.lpszClassName = TEXT("DirectXApplication_Tutorial"); //The TEXT() macro function is used to assure the projects compiles if UNICODE support is either enabled or disabled.
+```
+
+To note that a custom cursor, application icons and flags can be set in this structure.
+
+## 2.) Register the window class
+
+We register a window class using the RegisterClass function.
+
+```cpp
+if(RegisterClass(&wndClass) == 0) throw std::exception("Failed to register the window class!");; //registers the window class. A window class defines how the button behaves, i.e a button, or a label... We throw an exception if RegisterClass somehow fails...
```

Comment:
extraneous semicolon at the end,

Also do we want to throw exceptions?

Reply:
The comment should also probably be on the two lines before the conditional clause. That line is otherwise *extremely* long.

Reply:
> Also do we want to throw exceptions?

They can be caught later if for example we wrap the WINAPI funtion calls in a class. I find them more easier to understand compared to returning errors code, doing multiple checks, etc. Even if in many cases, exceptions are a sign of bad design.

Reply:
I think we have not discussed exceptions over return values to a full extent, but when we discussed yesterday during the meeting, that I personally would prefer return values in most cases, and when fail, most existing code std::cout to console for now, in the future shall be replaced by some logging lib...

Reply:
Well, I didn't join that meeting 😅.  Feel free to modify the code, I can't right now because I'm busy with some exams. Thanks

Reply:
Don't worry about it :) We will get this sorted out. and good luck with the exams

Reply:
Thank you! 👍